### PR TITLE
feat(services): revamp database accessibility field 

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-1262

2024-09-05-155520_684x452_scrot

2024-09-05-180658_834x382_scrot

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Help yourself at 2AM by putting some metrics to help observability

File: libs/pages/services/src/lib/ui/page-database-create/step-general/step-general.spec.tsx [hunk 1]
```diff
@@ -10,11 +11,26 @@ const props: StepGeneralProps = {
...
+    mode: DatabaseModeEnum.CONTAINER,
+    accessibility: DatabaseAccessibilityEnum.PUBLIC,
+    version: '12',
+    cpu: 512,
+    memory: 1024,
+  }
+
   it('should render successfully', () => {
-    const { baseElement } = render(wrapWithReactHookForm(<StepGeneral {...props} />))
+    const { baseElement } = render(
```
Comment (RemiBonnet) [inline libs/pages/services/src/lib/ui/page-database-create/step-general/step-general.spec.tsx:29]: using `renderWithProviders`

File: libs/pages/database/src/lib/ui/page-settings-general/page-settings-general.tsx [hunk 2]
```diff
@@ -166,30 +171,75 @@ export function PageSettingsGeneral({
...
+                        </SegmentedControl.Item>
+                      </SegmentedControl.Root>
+                      <p className="mt-2 text-sm text-neutral-350">
+                        {match({ watchMode, watchAccessibility })
+                          .with(
+                            { watchMode: 'CONTAINER', watchAccessibility: 'PRIVATE' },
+                            { watchMode: 'CONTAINER', watchAccessibility: undefined },
+                            () => (
+                              <>
+                                <strong>Private access to your database is ensured</strong>, as it is only accessible
```
Comment (RemiBonnet) [inline libs/pages/database/src/lib/ui/page-settings-general/page-settings-general.tsx:207]: it's not strong but `medium` should be a `<span className="font-medium">`

**Review by RemiBonnet — CHANGES_REQUESTED** [review]

File: libs/pages/services/src/lib/ui/page-database-create/step-general/step-general.spec.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/pages/services/src/lib/ui/page-database-create/step-general/step-general.spec.tsx]: fixed https://github.com/Qovery/console/pull/1651/commits/a11b475575c43e62269a4dd3cdc5bbef3bb6f020

File: libs/pages/database/src/lib/ui/page-settings-general/page-settings-general.tsx (hunk 2 above)
Comment (ctjhoa) [reply libs/pages/database/src/lib/ui/page-settings-general/page-settings-general.tsx]: fixed https://github.com/Qovery/console/pull/1651/commits/a11b475575c43e62269a4dd3cdc5bbef3bb6f020

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**